src/distances/mod.rs

```diff
@@ -334,3 +383,23 @@ pub fn self_dists_knn_precluster<'a>(
     }
     sp_distances
 }
+
+pub fn create_completeness_vector(
+    completeness_map: &HashMap<String, f64>,
+    sketches: &MultiSketch
+) -> Vec<f64> {
+    let mut completeness_vec = Vec::with_capacity(sketches.number_samples_loaded());
+    
+    for i in 0..sketches.number_samples_loaded() {
+        let genome_name = sketches.sketch_name(i);
+        let completeness = completeness_map.get(genome_name)
+            .copied()
+            .unwrap_or_else(|| {
+                log::warn!("Genome '{}' not found in completeness file, using default 1.0", genome_name);
```

Comment:
This could lead to a lot of messages if the file is wrong. I would suggest writing one message at the end with the full list of names (so store genome_name in a new vec of missing names, and print this message if it's longer than 0)

Reply:
did missing_genomes with capacity. I thought it might be better this way since we do not need to do extensive memory copies if we have many missing genomes. Or do you think this is overkill and we should not expect all the genomes to be missing all the time?